docs/changing-behavior-between-calls.md

```diff
@@ -17,17 +19,40 @@ A.CallTo(() => fakeShop.Address).Returns("123 Fake Street").NumberOfTimes(17);
 ```
 
 This could be useful if you want to allow a limited number of calls on
-a [strict fake](strict-fakes.md), but there's a more useful
-application.
+a [strict fake](strict-fakes.md), but there's a more useful application.
+
+## Specifying different behaviors for successive calls
+
+In some cases, you might want to specify different behaviors for successive
+calls to the same method. For instance, in order to test the System Under
+Test's retry logic, a Fake service could be configured to fail once and then
+function properly thereafter. This can be done by chaining behaviors like this:
+
+```csharp
+// Configure the method to throw an exception once, then succeed forever
+A.CallTo(() => fakeService.DoSomething())
+    .Throws<Exception>().Once()
+    .Then
+    .Returns("SUCCESS");
+```
+
+Note that you can only use `Then` after specifying a number of repetitions for
+the behavior.
+
 
-##Changing behavior between calls
+## Overriding the behavior for a call
 
 Call specifications act kind of like a stack - they're pushed on the
 Fake and then popped off once the number of repetitions defined for a
-call have been exhausted. Thus, it's possible to have a call to a Fake
-act one way, and then another. In order to test the System Under
-Test's retry logic, a Fake service could be configured to fail once
-and then function properly thereafter:
+call have been exhausted.
+
+This can be useful when you have a fake with a default behavior configured
+by a `FakeOptionsBuilder`, and you need to override this behavior for a
+specific test.
+
+Thus, it's possible to have a call to a Fake act one way, and then another. For
+instance, the same effect as the previous sample can be achieved by overriding
+the behavior of the fake:

```

Comment:
Love the reimplemented example. Very useful to confused people.

(In general, I like the all the docs changes. I think they'd be acceptable as is (except for the broken links I mentioned in [the conversation](https://github.com/FakeItEasy/FakeItEasy/pull/804)), but I'm trying to be super picky just in case we can alleviate confusion for even one person.